modules/internal/DefaultRectangular.chpl

```diff
@@ -705,6 +716,164 @@ module DefaultRectangular {
+
+          if blkChanged {
+            for param i in 1..rank {
+              sum += ind(i) * blk(i);
+            }
+          } else {
+            for param i in 1..rank-1 {
+              sum += ind(i) * blk(i);
+            }
+            sum += ind(rank);
+          }
+
+          if !earlyShiftData then sum -= factoredOffs;
+          if chunkify then
+            return chunked_dataIndex(sum);
+          else
+            return sum;
+        }
+      }
+  }
+
+  proc _remoteAccessData.computeFactoredOffs() {
+    for param i in 1..rank do {
+      factoredOffs = factoredOffs + blk(i) * off(i);
+    }
+  }
+
+  proc _remoteAccessData.initShiftedData() {
+    if earlyShiftData && !stridable {
+      type idxSignedType = chpl__signedType(idxType);
+      const shiftDist = if isIntType(idxType) then origin - factoredOffs
+                        else origin:idxSignedType - factoredOffs:idxSignedType;
+      shiftedData = _ddata_shift(eltType, data, shiftDist);
+    }
+  }
+
+  proc _remoteAccessData.toSlice(newDom) {
+    compilerAssert(defRectSimpleDData && this.rank == newDom.rank);
+    var rad : _remoteAccessData(eltType, newDom.rank, newDom.idxType, newDom.stridable, newDom.stridable || this.blkChanged);
+
+    rad.data        = this.data;
+    rad.shiftedData = if newDom.stridable then this.data else this.shiftedData;
+    rad.origin      = this.origin:newDom.idxType;
+    rad.off         = chpl__tuplify(newDom.dsiLow);
+    rad.str         = chpl__tuplify(newDom.dsiStride);
+
+    for param i in 1..rank {
+      const shift = this.blk(i) * (newDom.dsiDim(i).low - this.off(i)) / abs(this.str(i)) : rad.idxType;
+      if this.str(i) > 0 {
+        rad.origin += shift;
+      } else {
+        rad.origin -= shift;
+      }
+
+      const mult = (newDom.dsiDim(i).stride / this.str(i)) : rad.idxType;
+      rad.blk(i) = this.blk(i) * mult;
+    }
+
+    rad.computeFactoredOffs();
+    rad.initShiftedData();
+
+    return rad;
+  }
+
+  proc _remoteAccessData.toReindex(newDom) {
+    compilerAssert(defRectSimpleDData && this.rank == newDom.rank);
+    var rad : _remoteAccessData(eltType, newDom.rank, newDom.idxType, newDom.stridable, blkChanged);
+
+    rad.data        = this.data;
+    rad.shiftedData = if newDom.stridable then this.data else this.shiftedData;
+    rad.origin      = this.origin:newDom.idxType;
+    rad.blk         = this.blk;
+    rad.off         = chpl__tuplify(newDom.dsiLow);
+    rad.str         = chpl__tuplify(newDom.dsiStride);
+    rad.factoredOffs = 0:idxType;
+
+    rad.computeFactoredOffs();
+    rad.initShiftedData();
+
+    return rad;
+  }
+
+  proc _remoteAccessData.toRankChange(newDom, cd, idx) {
+    compilerAssert(defRectSimpleDData && this.rank == idx.size && this.rank != newDom.rank);
+    // TODO: If 'collapsedDims' were param, we would know if blk(rank) was 1 or not.
+    var rad : _remoteAccessData(eltType, newDom.rank, newDom.idxType, newDom.stridable, true);
```

Comment:
I think it'd be nice to call out the places that set this in comments. When I was trying to understand that field, I searched for blkChanged, and this one at least isn't findable that way.